### PR TITLE
Module2; week2- To do list interactive

### PR DESCRIPTION
**Pull request by uno36**
I made the following modifications to the code:

- [ ] Added an event listener to the checkbox that listens for the 'change' event.
- [ ] Updated the items object's completed value to reflect the user's actions upon checking or unchecking the checkbox.
- [ ] Implemented a new function named clearAllTasks that removes all the tasks and their associated horizontal rule elements from the to-do list.
- [ ] Added an event listener to the "Clear all completed" button that calls the clearAllTasks function when clicked.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  tasks.sort((a, b) => a.index - b.index);
+  todoList.innerHTML = tasks
+    .map(
+      (task) => `<li class="todo-item ${task.completed ? 'completed' : ''}">
+          <input type="checkbox" ${task.completed ? 'checked' : ''}>
+          <span>${task.description}</span>
+          <hr class="todo-hr">
+        </li>`,
+    )
+    .join('');
```
Comment (AnselemOdims) [inline src/index.js:16]: - [ ] At the moment, when a user lands on the page they see an `undefined` todo item. Please kindly fix this.
- [ ] Also, when the page is reloaded, all added todo items get removed from the screen and localStorage. Please kindly fix that

File: src/functionality.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      if (hr && hr.tagName === 'HR') {
+        todoList.removeChild(hr);
+      }
+    }
+  }
+  saveTasksToLocalStorage();
+};
+
+const clearButton = document.getElementById('submit');
+clearButton.addEventListener('click', clearCompletedTasks);
```
Comment (AnselemOdims) [inline src/functionality.js:167]: - [ ] The clear all completed only removes the items from the screen and not the localstorage. Please kindly ensure that when the clear all completed button is clicked, all completed items are removed from both the screen and the localstorage

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @uno36 , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/functionality.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  const storedTasks = JSON.parse(localStorage.getItem('tasks')) || [];
+  tasks.length = 0;
+  tasks.push(...storedTasks);
+  renderTodoList();
+}
+loadTasksFromLocalStorage();
+
+clearButton.addEventListener('click', () => {
+  localStorage.clear();
+});
```
Comment (braincee) [inline src/functionality.js:179]: - [ ] Please as suggested by the previous reviewer, the `Clear All Completed` button only removes data from the screen and not from the local storage. The data should not be persistent but removed from the local storage when the button is clicked. 😄

**Review by braincee — CHANGES_REQUESTED** [review]
Hello @uno36 👋 , I am @braincee your Code Reviewer

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


so-close

HIGHLIGHTS
------

 - GitHub flow rightfully used ✅
 - Meaningful title and summary for PR ✅
 - No linter errors ✅

REQUIRED CHANGES ♻️
------
**_Please check the comments under the review._**

OPTIONAL CHANGES
--------
**_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @braincee in your question so I can receive the notification.

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

----

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @uno36 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- All requested changes implemented ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**